crates/codegen/parser/runtime/src/support/choice_helper.rs

```diff
@@ -1,8 +1,12 @@
 use std::mem;
 use std::ops::ControlFlow;
 
-use super::{context::Marker, ParserContext, ParserResult};
-use crate::{cst, kinds::TokenKind, parse_error::ParseError, text_index::TextIndex};
+use super::context::Marker;
```

Comment:
since all of our imports are nicely grouped and sorted now, should we allow `super::` in addition to `crate::`?
I would like to standardize on `crate::` unless it is actually not possible, which is not the case here.

Thinking out loud. Not blocking for this PR.

Reply:
Using up to one `super` in a deeper (not direct-child-to-root) module is acceptable, that's what it's for.

It'd be less readable/understandable IMO if we had to always to through the root; _especially_ if the item from the parent module is private to that module and children, since the access to the item, in terms of privacy rules, comes from the child and not its crate root.

Reply:
I think I'm going to use your arguments above here: 😅 

- it naturally settles on the module as the, well, module/boundary from which items can be imported... thus, helps build the intuition how to use and structure Rust crates.
- repeats enough information to quickly trace a module path rather than having to reconstruct the paths from the current file path...
- quite good greppability

Reply:
I don't even mean sibling modules `use super::sibling::something` but direct parent `use super::something`.

These answers from the thread resonate with me (it's recent, not that long, and has answers from few notable community members): https://www.reddit.com/r/rust/comments/15km47h/which_import_syntax_to_use_use_cratemodule_or_use/

> I exclusively use super for tests and other internal modules that solely depend on the parent. But otherwise I stick with crate to make it easier to move things around. 

> In general, I tend to only use super:: for imports of things that live directly in the parent module.

As such, I'm happy to change the `rust-analyzer.imports.prefix` option to `crate`, so all new imports always begin with crate; however, it makes sense to still use `super`-imports in inline modules (e.g. for tests or grouping) or to pull stuff from direct parent whenever needed.

Since there's no Clippy lint for that and Rustfmt can't convert/enforce style of imports (it'd need semantic privacy knowledge) we just have to rely on reasonable auto-imports and the reasonable programmer to maintain this style :sweat_smile:

Reply:
Changed `super`-style imports to `crate` whenever possible with the exception of inline modules :+1: